gammapy/modeling/models/spatial.py

```diff
@@ -45,6 +52,34 @@ def position(self):
         except IndexError:
             raise ValueError("Model does not have a defined center position")
 
+    _position_error = None
+
+    @property
+    def position_error(self):
+        """Set position error from covariance matrix"""
+        if self.parameters.covariance is None:
+            return self._position_error
+        else:
```

Comment:
Suggest to remove the `else:` and de-dent this.
I.e. handle the abnormal case first and return early.
And then have the main code at the left-most indentation level.